scripts/python/app.py

```diff
@@ -0,0 +1,600 @@
+#!/usr/bin/env python3
+# =========================================================================================
+# (C) (or copyright) 2020-2021. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+# =========================================================================================
+
+import os
+import datetime
+import filecmp
+import pathlib
+import json
+import shutil
+import base64
+from io import BytesIO
+import jwt
+import pem
+import pycurl
+from git import Repo
+import git
+
+class Node:
+
+    """
+    Class used to store branch contents in terms of files, directories or misc
+    """
+
+    def __init__(self, dir_name="", rel_path=""):
+        """Creating a Node object
+
+        dir_name is the name of the directory the node contains information about 
+        rel_path is the actual path to the directory
+        """
+
+        self.dir = dir_name
+        self.dirs = []
+        self.files = []
+        self.misc = []
+        self.rel_path = rel_path + dir_name
+
+    def insert(self, content, content_type):
+        """Record the contents of a directory by inserting it
+
+        Will either store new information as a file, directory or misc type.
+        If the content type is of type dir than a new node is created.
+        """
+        if content_type == "dir":
+            self.dirs.append(Node(content, self.rel_path + "/"))
+        elif content_type == "file":
+            self.files.append(content)
+        else:
+            self.files.append(content)
+
+    def getNodes(self):
+        """Returns a list of all nodes in the current node, which are essentially
+        directories."""
+        return self.dirs
+
+    def getPath(self):
+        """Get the relative path of the current node."""
+        return self.rel_path
+
+    def printTree(self):
+        """Print contents of node and all child nodes"""
+        print("Contents in folder: " + self.rel_path)
+        for fil in self.files:
+            print("File " + fil)
+        for mis in self.misc:
+            print("Misc " + mis)
+        for node in self.dirs:
+            node.printTree()
+
+
+
+class App:
```

Comment:
The name `App` is not very descriptive. We may want to think of a different name. Something like "PerformanceReporter" maybe?

Reply:
So I did actually have a reason for this. I have opened an EIDR trying to get the App file open sourced as a standalone item. It is meant to be generic. The actual implementation of App is in the parthenon_metrics_app.py that is directly relevant to parthenon. In all actuallity the App class could be used to create several different apps. If you want something a little more specific I could change it to GithubApp but beyond that I'm not sure what else would be appropriate.

Reply:
I would prefer `GitHubApp`. `App` is just too generic---my initial instinct would be to read the name as "oh this is the parthenon application," which is confusing.